wifiphisher/common/firewall.py

```diff
@@ -1,46 +1,60 @@
+from wifiphisher.common.constants import (NETWORK_GW_IP, SSL_PORT, PORT, DN)
 
 
 class Fw():
-    def __init__(self):
-        pass
+    """Handle iptable routing."""
 
     def nat(self, internal_interface, external_interface):
-        subprocess.call(
-            ('iptables -t nat -A POSTROUTING -o %s -j MASQUERADE' %
-             (external_interface, )),
-            shell=True)
-
-        subprocess.call(
-            ('iptables -A FORWARD -i %s -o %s -j ACCEPT' %
-             (internal_interface, external_interface)),
-            shell=True)
+        # type: (str, str) -> None
+        """Create NAT in iptalbes."""
+        execute_commands([
+            "iptables -t nat -A POSTROUTING -o {} -j MASQUERADE".format(
+                external_interface),
+            "iptables -A FORWARD -i {} -o {} -j ACCEPT".format(
+                internal_interface, external_interface)
+        ])
 
     def clear_rules(self):
-        subprocess.call('iptables -F', shell=True)
-        subprocess.call('iptables -X', shell=True)
-        subprocess.call('iptables -t nat -F', shell=True)
-        subprocess.call('iptables -t nat -X', shell=True)
+        # type: () -> None
+        """Reset all the routing."""
+        execute_commands([
+            "iptables -F", "iptables -X", "iptables -t nat -F",
+            "iptables -t nat -X"
+        ])
 
     def redirect_requests_localhost(self):
-        subprocess.call(
-            ('iptables -t nat -A PREROUTING -p tcp --dport 80 -j DNAT --to-destination %s:%s'
-             % (NETWORK_GW_IP, PORT)),
-            shell=True)
-        subprocess.call(
-            ('iptables -t nat -A PREROUTING -p udp --dport 53 -j DNAT --to-destination %s:%s'
-             % (NETWORK_GW_IP, 53)),
-            shell=True)
-        subprocess.call(
-            ('iptables -t nat -A PREROUTING -p tcp --dport 53 -j DNAT --to-destination %s:%s'
-             % (NETWORK_GW_IP, 53)),
-            shell=True)
-        subprocess.call(
-            ('iptables -t nat -A PREROUTING -p tcp --dport 443 -j DNAT --to-destination %s:%s'
-             % (NETWORK_GW_IP, SSL_PORT)),
-            shell=True)
+        # type: () -> None
+        """Redirect all the request to local host."""
+        execute_commands([
+            "iptables -t nat -A PREROUTING -p tcp --dport 80 -j DNAT"
+            "--to-destination {}:{}".format(NETWORK_GW_IP, PORT),
+            "iptables -t nat -A PREROUTING -p udp --dport 53 -j DNAT"
+            "--to-destination {}:{}".format(NETWORK_GW_IP, 53),
+            "iptables -t nat -A PREROUTING -p tcp --dport 53 -j DNAT"
+            "--to-destination {}:{}".format(NETWORK_GW_IP, 53),
+            "iptables -t nat -A PREROUTING -p tcp --dport 443 -j DNAT"
+            "--to-destination {}:{}".format(NETWORK_GW_IP, SSL_PORT)
+        ])
 
     def on_exit(self):
+        # type: () -> None
+        """Start the clean up."""
         self.clear_rules()
+
+
+def execute_commands(commands):
```

Comment:
This doesn't look like something that belongs here. Maybe create a `utils.py` file and put it there.